libcloud/storage/drivers/s3.py

```diff
@@ -1015,23 +1015,50 @@ class S3SAEastStorageDriver(S3StorageDriver):
     ex_location_name = 'sa-east-1'
 
 
-class S3RGWOutscaleConnection(S3Connection):
+class S3RGWOutscaleConnectionAWS4(SignedAWSConnection, BaseS3Connection):
+    service_name = 's3'
+    version = API_VERSION
+
+    def __init__(self, user_id, key, secure=True, host=None, port=None,
+                 url=None, timeout=None, proxy_url=None, token=None,
+                 retry_delay=None, backoff=None):
+
+        super(S3RGWOutscaleConnectionAWS4, self).__init__(user_id, key,
+                                                          secure, host,
+                                                          port, url,
+                                                          timeout,
+                                                          proxy_url, token,
+                                                          retry_delay,
+                                                          backoff,
+                                                          4)  # force aws4
+
+
+class S3RGWOutscaleConnectionAWS2(S3Connection):
     pass
 
 
 class S3RGWOutscaleStorageDriver(S3StorageDriver):
 
     def __init__(self, key, secret=None, secure=True, host=None, port=None,
                  api_version=None, region=S3_RGW_OUTSCALE_DEFAULT_REGION,
-                 **kwargs):
+                 signature_version=DEFAULT_SIGNATURE_VERSION, **kwargs):
         if region not in S3_RGW_OUTSCALE_HOSTS_BY_REGION:
             raise LibcloudError('Unknown region (%s)' % (region), driver=self)
         self.name = 'OUTSCALE Ceph RGW S3 (%s)' % (region)
         self.ex_location_name = region
         self.region_name = region
-        self.connectionCls = S3RGWOutscaleConnection
-        self.connectionCls.host = S3_RGW_OUTSCALE_HOSTS_BY_REGION[region]
+        self.signature_version = signature_version
+        self.host = S3_RGW_OUTSCALE_HOSTS_BY_REGION[region]
         super(S3RGWOutscaleStorageDriver, self).__init__(key, secret,
                                                          secure, host, port,
                                                          api_version, region,
                                                          **kwargs)
+
+    def _ex_connection_class_kwargs(self):
```

Comment:
@kami Thanks for the review. It is very appreciated.

Yes, please. Feel free to merge. We can continue improving the code over the upstream version. I guess my next patch would build on a similar bug so they could be fixed together.

I guess the example code I used at https://github.com/apache/libcloud/pull/736#issuecomment-218434459 is not catching the issue properly although it looks working ok? I am manipulating the connectionCls inside the driver constructor cause it looks the only place to choose between the two different connection classes without a further modification.